content/blog/2024-09-10-memory-access-amortization.md

```diff
@@ -0,0 +1,158 @@
+
+
+### How to reproduce Valkey 8.0 performance numbers 
+
+This section will walk you through the process of reproducing our performance results, where we achieved 1.19 million requests per second using Valkey 8.
+
+### Hardware Setup
+
+We conducted our tests on an AWS EC2 c7g.4xlarge instance, featuring 16 cores on an ARM-based (aarch64) architecture.
+
+### System Configuration
+
+> Note: The core assignments used in this guide are examples. Optimal core selection may vary depending on your specific system configuration and workload.
+
+Interrupt affinity - locate the network interface with `ifconfig` (let's assume it is `eth0`) and its associated IRQs with 
+<pre class="inverted-code">
+grep eth0 /proc/interrupts | awk '{print $1}' | cut -d : -f 1
+</pre>
+In our setup, lines `48` to `55` are allocated for `eth0` interrupts. Allocate one core per 4 IRQ lines: 
+<pre class="inverted-code">
+for i in {48..50}; do echo 1000 > /proc/irq/$i/smp_affinity; done
+for i in {51..55}; do echo 2000 > /proc/irq/$i/smp_affinity; done  
+</pre>
+Server configuration - launch the Valkey server with these minimal configurations:
+<pre class="inverted-code">
+./valkey-server --io-threads 9 --save --protected-mode no
+</pre>
+`--save` disables dumping to RDB file and `--protected-mode no `  allows connections from external hosts. `--io-threads` number includes the main thread and the IO threads, meaning that in our case 8 I/O threads are launched in addition to the main thread. 
+
+Main thread affinity - pin the main thread to a specific CPU core, avoiding the cores handling IRQs. Here we use core #3:
+<pre class="inverted-code">
+sudo taskset -cp 3 `pidof valkey-server`
+</pre>
+> Important: We suggest to experiment with different other to find the optimal performance while avoiding conflicts with the IRQ-handling cores.
+
+### Benchmark Configuration 
+
+Run the benchmark from a separate instance using the following parameters:
+
+* Value size: 512 bytes
+* Number of keys: 3 million
+* Number of clients: 650
+* Number of threads: 50 (may vary for optimal results)
+
+<pre class="inverted-code">
+./valkey-benchmark -t set -d 512 -r 3000000 -c 650 --threads 50 -h <host-name> -n 100000000000
```

Comment:
`<host-name>` is getting rendered out, so need some other way to indicate it without <>

Reply:
&lt;host-name&gt; might work

Reply:
`&lt;host-name&gt; ` does the job. Thanks

Reply:
If you use the backticks, you don't have to do anything special, it won't think `<host-name>` is HTML

<img width="555" alt="Screenshot 2024-09-11 at 1 51 00 PM" src="https://github.com/user-attachments/assets/da60375c-4397-4df9-8872-88dd847d0a58">

Which looks correct:

![Screenshot 2024-09-11 at 1 51 23 PM](https://github.com/user-attachments/assets/6499e1a0-990d-445c-a1a7-1cd4fdf738fe)